src/common/scratch_buffer.hpp

```diff
@@ -1,6 +1,8 @@
-// scratch_buffer.h
-
+#pragma once
 #include <inttypes.h>
+#include <stddef.h>
+
+namespace buddy::scratch_buffer {
 
 /// Single place for definition of large RAM space for various tasks.
 /// Currently, there's no locking mechanism so before you use it, make sure
```

Comment:
... aren't you just adding some locking mechanizm :wink: ? Please update the comment accordingly as well.

Reply:
Nice catch ☺️